ContactsManagerBackend/ContactsManager/ContactsManagerDAL/PhonesProvider.cs

```diff
@@ -0,0 +1,148 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Data.SqlClient;
+using System.Linq;
+using ContactsManagerDAL.Models;
+using DBExtentionsMethods;
+
+namespace ContactsManagerDAL
+{
+    public class PhonesProvider
+    {
+        public List<Phone> GetPhones(Guid contactId)
+        {
+            List<Phone> Phones = new List<Phone>();
+
+            using (SqlConnection conn = DBConnection.GetSqlConnection())
+            using (SqlCommand cmd = conn.CreateCommand())
+            {
+                cmd.CommandText = "getContactPhones";
+                cmd.CommandType = CommandType.StoredProcedure;
+                cmd.Parameters.AddNewParameter("ContactId", SqlDbType.UniqueIdentifier, contactId);
+
+                var reader = cmd.ExecuteReader(CommandBehavior.CloseConnection);
+
+                if (reader.HasRows)
+                {
+                    while (reader.Read())
+                    {
+                        var phone = new Phone();
+
+                        phone.LoadDataFromReader(reader);
+
+                        Phones.Add(phone);
+                    }
+                }
+            }
+
+            return Phones;
+        }
+
+        public Phone GetPhone(Guid contactId, Guid Id)
+        {
+            var phone = new Phone();
+
+            using (SqlConnection conn = DBConnection.GetSqlConnection())
+            using (SqlCommand cmd = conn.CreateCommand())
+            {
+                cmd.CommandText = "getContactPhoneById";
+                cmd.CommandType = CommandType.StoredProcedure;
+
+                cmd.Parameters.AddNewParameter("Id", SqlDbType.UniqueIdentifier, Id);
+                cmd.Parameters.AddNewParameter("ContactId", SqlDbType.UniqueIdentifier, contactId);
+
+                var reader = cmd.ExecuteReader(CommandBehavior.CloseConnection);
+
+                if (reader.Read())
+                {
+                    phone.LoadDataFromReader(reader);
+                }
+                else
+                {
+                    phone = null;
```

Comment:
Consider making the LoadDataFromReader methods static construction methods, i.e. they should return the new object.
That way you'll avoid making an object needlessly. 
Alternatively you could start with null and create the object in "if (reader.Read())" block